packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -34,6 +34,179 @@ export interface HttpRouteMatch {
    * and you want the route to match requests to my-service.local/metrics, your prefix should be /metrics.
    */
   readonly prefixPath: string;
+
+  /**
+   * Specifies the client request headers to match on
+   * @default - do not match on headers
+   */
+  readonly headers?: HttpRouteHeader[];
+
+  /**
+   * The client request method to match on. Specify only one.
+   * @default - do not match on request method
+   */
+  readonly method?: HttpRouteMatchMethod;
+
+  /**
+   * The client request scheme to match on. Specify only one. Applicable only for HTTP2 routes.
+   * @default - do not match on HTTP2 scheme
+   */
+  readonly scheme?: HttpRouteMatchScheme;
+}
+
+/**
+ * Supported request matching methods.
+ */
+export enum HttpRouteMatchMethod {
+  /**
+   * GET request
+   */
+  GET = 'GET',
+
+  /**
+   * HEAD request
+   */
+  HEAD = 'HEAD',
+
+  /**
+   * POST request
+   */
+  POST = 'POST',
+
+  /**
+   * PUT request
+   */
+  PUT = 'PUT',
+
+  /**
+   * DELETE request
+   */
+  DELETE = 'DELETE',
+
+  /**
+   * CONNECT request
+   */
+  CONNECT = 'CONNECT',
+
+  /**
+   * OPTIONS request
+   */
+  OPTIONS = 'OPTIONS',
+
+  /**
+   * TRACE request
+   */
+  TRACE = 'TRACE',
+
+  /**
+   * PATCH request
+   */
+  PATCH = 'PATCH',
+}
+
+/**
+ * Supported :scheme options for HTTP2
+ */
+export enum HttpRouteMatchScheme {
+  /**
+   * Match HTTP requests
+   */
+  HTTP = 'http',
+
+  /**
+   * Match HTTPS requests
+   */
+  HTTPS = 'https',
+}
+
+/**
+ * An object that represents the HTTP header in the request to match.
+ */
+export interface HttpRouteHeader {
+  /**
+   * Specify `true` to match anything except the match criteria.
+   * @default false
+   */
+  readonly invert?: boolean;
+
+  /**
+   * The method to use to match the header.
+   * @default - no match method
+   */
+  readonly match?: IHeaderMatchMethod;
+
+  /**
+   * A name for the HTTP header in the client request that will be matched on.
+   * @example 'content-type'
+   */
+  readonly name: string;
+}
```

Comment:
I gotta say, I'm not a huge fan of this API. Feels a little bit like too much ceremony for something simple. Here's how I would envision an API that I would like:

```ts
router.addRoute('route-http2', {
  routeSpec: appmesh.RouteSpec.http2({
    weightedTargets: [
      {
        virtualNode: node,
      },
    ],
    match: {
      prefixPath: '/my/path',
      headers: [
        appmesh.HeaderMatch.valueIs('Content-Type', 'application/json'),
        appmesh.HeaderMatch.valueStartsWith('Content-Type', 'application/json'),
        appmesh.HeaderMatch.valueEndsWith('Content-Type', 'application/json'),
        appmesh.HeaderMatch.valueMatchesRegex('Content-Type', 'application/.*'),
        appmesh.HeaderMatch.valueIsBetween('Content-Type', 1, 5),
        appmesh.HeaderMatch.valueIsNot('Content-Type', 'application/json'),
        appmesh.HeaderMatch.valueDoesNotStartWith('Content-Type', 'application/json'),
        appmesh.HeaderMatch.valueDoesNotEndWith('Content-Type', 'application/json'),
        appmesh.HeaderMatch.valueDoesNotMatchRegex('Content-Type', 'application/.*'),
        appmesh.HeaderMatch.valueIsNotBetween('Content-Type', 1, 5),
      ],
    },
  }),
});
```

What do you think of this idea @misterjoshua ?

Reply:
@skinny85 This looks pretty good! In this PR, opted for the way @dfezzie suggested in #13350 only because I didn't have a strong enough opinion, so I'm easily convinced. Lol. I have enough to get started.

nit., I'm not sure about `valueIsBetween` because the range App Mesh wants is `[start, end)` (inclusive of start, exclusive of end) and "Between" has some linguistic closed interval baggage. With this in mind, how about `valueIsInRange`?

Reply:
Another point about the `headers` property: I assume the result is a logical AND of all of the conditions inside of the array? If so, it might be worth it to call it out explicitly in the documentation.

Reply:
Ok. I've pushed up a docs change that mentions this.